sw/device/tests/sim_dv/lc_walkthrough_test.c

```diff
@@ -44,19 +46,106 @@ static volatile const uint8_t kOtpUnlockToken[LC_TOKEN_SIZE] = {
     0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
 };
 
+// LC rma token value for LC state transition.
+static volatile const uint8_t kLcRmaToken[LC_TOKEN_SIZE] = {
+    0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+    0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+};
+
+// LC rma token value in OTP secret2 partition.
+static volatile const uint8_t kOtpRmaToken[LC_TOKEN_SIZE] = {
+    0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+    0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+};
+
+static void lock_otp_secret0_partition() {
+  uint64_t otp_unlock_token_0 = 0;
+  uint64_t otp_unlock_token_1 = 0;
+  for (int i = 0; i < LC_TOKEN_SIZE; i++) {
+    if (i < LC_TOKEN_SIZE / 2) {
+      otp_unlock_token_0 |= (uint64_t)kOtpUnlockToken[i] << (i * 8);
+    } else {
+      otp_unlock_token_1 |= (uint64_t)kOtpUnlockToken[i]
+                            << ((i - LC_TOKEN_SIZE / 2) * 8);
+    }
+  }
```

Comment:
Thanks Alphan for the detailed suggestions. I will fix it soon.

Reply:
Hey @alphan I actually got a compile error here. I want to check with you:
`kOtpUnlockToken` is a volatile type, do I need to do some casting to make it work for `memcpy`?
Looks like the compile error is like this:
```
../mnt/repo_top/sw/device/tests/sim_dv/lc_walkthrough_test.c:64:31: error: passing 'const volatile uint8_t (*)[16]' to parameter of type 'const void *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
```

Reply:
Ah, apologies for missing this. `memcpy` won't work if these are `volatile`. If you want, you can use a for loop (like the ones already used in this RP) instead of `memcpy` though.

Reply:
I see! Thanks Alphan for clarifying that. Was a good learning case for me hehe.
Do you mind if I merge this today and do that in a separate PR? I think there are a few other tests that will benefit from this utility function, and I will do a clean up.

Reply:
Sure, no need to block this PR.

Reply:
https://github.com/lowRISC/opentitan/issues/12827 documented here. thanks Alphan.